File: src/main/java/no/nav/syfo/controller/domain/sykepengesoknad/oppsummering/Sporsmal.java
package no.nav.syfo.controller.domain.sykepengesoknad.oppsummering;

import java.io.Serializable;
import java.util.List;

import static java.util.Collections.emptyList;

public class Sporsmal implements Serializable {
    public Ledetekst ledetekst;
    public List<Svar> svar = emptyList();
    public Sporsmalstype type;

    public Sporsmal withLedetekst(final Ledetekst ledetekst) {
        this.ledetekst = ledetekst;
        return this;
    }

    public Sporsmal withSvar(final List<Svar> svar) {
        this.svar = svar;
        return this;
    }

    public Sporsmal withType(final Sporsmalstype type) {
        this.type = type;
        return this;
    }
}

File: src/main/kotlin/no/nav/syfo/controller/domain/sykepengesoknad/oppsumering/Sporsmal.kt
package no.nav.syfo.controller.domain.sykepengesoknad.oppsumering

import java.io.Serializable

class Sporsmal : Serializable {
    var ledetekst: Ledetekst? = null
    var svar: List<Svar> = emptyList()
    var type: Sporsmalstype? = null
    fun withLedetekst(ledetekst: Ledetekst?): Sporsmal {
        this.ledetekst = ledetekst
        return this
    }

    fun withSvar(svar: List<Svar>): Sporsmal {
        this.svar = svar
        return this
    }

    fun withType(type: Sporsmalstype?): Sporsmal {
        this.type = type
        return this
    }
}
